bundles/org.openhab.core.config.discovery/src/main/java/org/openhab/core/config/discovery/AbstractDiscoveryService.java

```diff
@@ -454,4 +439,49 @@ private boolean getAutoDiscoveryEnabled(Object autoDiscoveryEnabled) {
     private String inferKey(DiscoveryResult discoveryResult, String lastSegment) {
         return "discovery." + discoveryResult.getThingUID().getAsString().replaceAll(":", ".") + "." + lastSegment;
     }
+
+    protected DiscoveryResult getLocalizedDiscoveryResult(final DiscoveryResult discoveryResult,
```

Comment:
This now looks like an important method which has no test coverage. Do you mind adding some tests? Would be very much appreciated.

Reply:
Unit tests added